crates/dbsp/src/operator/z1.rs

```diff
@@ -199,24 +208,69 @@ pub struct Z1<T> {
     zero: T,
     empty_output: bool,
     values: T,
+    persistent_id: String,
+}
+
+#[derive(rkyv::Serialize, rkyv::Deserialize, rkyv::Archive)]
+pub struct CommittedZ1 {
+    values: Vec<u8>,
+}
+
+impl<T> From<&Z1<T>> for CommittedZ1
+where
+    T: Checkpoint + Clone,
+{
+    fn from(z1: &Z1<T>) -> CommittedZ1 {
+        CommittedZ1 {
+            values: z1.values.checkpoint(),
+        }
+    }
 }
 
 impl<T> Z1<T>
 where
-    T: Clone,
+    T: Checkpoint + Clone,
 {
-    pub fn new(zero: T) -> Self {
-        Self {
-            zero: zero.clone(),
-            empty_output: false,
-            values: zero,
+    pub fn new(persistent_id: String, zero: T) -> Self {
+        if let Some(cid) = Runtime::restore_from_commit() {
+            let z1_path = Self::checkpoint_file(cid, &persistent_id);
+            let content = fs::read(z1_path).expect("Z1 meta-data for checkpoint must exist.");
```

Comment:
I guess, you're trying to avoid even bigger API restructuring by panicking instead of failing. Shall we file an issue to fix this eventually and bubble up errors from checkpoint/restarts?

Reply:
Yes this would lead to many more changes, but it's not super critical in my mind since it happens on circuit initialization, s if this panics it will panic very early before we do anything useful.